photon-api/src/main/scala/com/linkedin/photon/ml/algorithm/RandomEffectCoordinate.scala

```diff
@@ -299,40 +292,40 @@ object RandomEffectCoordinate {
               val updatedModel = optimizationProblem.run(trainingLabeledPoints, localModel)
               val stateTrackers = optimizationProblem.getStatesTracker
 
-              (updatedModel, stateTrackers)
+              (updatedModel, Some(stateTrackers))
 
             case (localModel, _) =>
               (localModel, None)
           }
+        modelsAndTrackers.persist(StorageLevel.MEMORY_ONLY_SER)
+
+        val models = modelsAndTrackers.mapValues(_._1)
+        val optimizationTracker = RandomEffectOptimizationTracker(modelsAndTrackers.flatMap(_._2._2))
```

Comment:
Two methods to access elements in a tuple: 1) using underscore + index; 2) using assigned names to elements. Which one is preferred in Photon ML? I feel using underscore is less readable.

Reply:
I think it depends on the situation and the name of the tuple. If you feel it's clear what the type of the anonymous value is, then using the underscore is fine. If you feel that it benefits from additional information, then a named parameter is better. Do you think it's unclear in this situation? I thought the name `modelsAndTrackers` for the parameter makes the types in the tuple clear.

Reply:
The name `modelsAndTrackers` makes the type clear, but it's not obvious for me of the second `_2` right after the track. Actually I am okay with both, as you already kept it quite consistent.